docs/new-client.md

```diff
@@ -0,0 +1,76 @@
+# Writing a new editor client
+
+Before writing a new editor client, first check if someone else has managed to
+integrate Metals with your favorite text editor.
+
+* [VS Code](https://github.com/scalameta/metals/blob/master/vscode-extension/src/extension.ts),
+  maintained in this repo
+* [Atom](https://atom.io/packages/ide-scala), maintained by
+  [@laughedelic](https://github.com/laughedelic)
+* [Emacs](https://github.com/rossabaker/lsp-scala), maintained by
+  [@rossabaker](https://github.com/rossabaker)
+* Others, see [#217](https://github.com/scalameta/metals/issues/217)
+
```

Comment:
Suggest here to open an issue/PR for any new client implementations not mentioned here yet?